builder/qemu/step_run_test.go

```diff
@@ -0,0 +1,148 @@
+package qemu
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/packer/helper/multistep"
+	"github.com/hashicorp/packer/packer"
+	"github.com/stretchr/testify/assert"
+)
+
+func runTestState(t *testing.T, config *Config) multistep.StateBag {
+	state := new(multistep.BasicStateBag)
+
+	state.Put("ui", packer.TestUi(t))
+	state.Put("config", config)
+
+	d := new(DriverMock)
+	d.VersionResult = "3.0.0"
+	state.Put("driver", d)
+
+	state.Put("commHostPort", 5000)
+	state.Put("floppy_path", "fake_floppy_path")
+	state.Put("http_ip", "127.0.0.1")
+	state.Put("http_port", 1234)
+	state.Put("iso_path", "/path/to/test.iso")
+	state.Put("qemu_disk_paths", []string{})
+	state.Put("vnc_port", 5905)
+	state.Put("vnc_password", "fake_vnc_password")
+
+	return state
+}
+
+func Test_getCommandArgs(t *testing.T) {
+	state := runTestState(t, &Config{})
+
+	args, err := getCommandArgs("", state)
```

Comment:
Nice, getCommandArgs has sooo many cases; it's great to test it 👍

Reply:
This doesn't even begin to cover them all but it's a start and at least covers the cd_files option I added last week